src/main/java/com/neprozorro/service/LotInfoService.java

```diff
@@ -93,18 +93,21 @@ private Predicate inOrEqualLotStatus(Root<LotInfo> root, CriteriaBuilder criteri
         return root.get(column).in(lotStatuses);
     }
 
-    private Predicate betweenOrEqual(Root<LotInfo> root, CriteriaBuilder criteriaBuilder, String column, String value) {
+    private Predicate rangeTotalPrice(Root<LotInfo> root, CriteriaBuilder criteriaBuilder, String column, String value) {
         String[] parsedValue = value.split(",");
 
-        if (parsedValue.length == 1) {
-            return criteriaBuilder.equal(root.get(column), BigDecimal.valueOf(Double.valueOf(value)));
-        }
+        BigDecimal minValue = BigDecimal.valueOf(Double.parseDouble(parsedValue[0]));
+        BigDecimal maxValue = BigDecimal.valueOf(Double.parseDouble(parsedValue[1]));
+
+        if (parsedValue.length != 2 ) throw new RuntimeException("Invalid input format");
```

Comment:
0 and 1 are fine but other numbers are considered "magic numbers" when they are used in the code this way. You should create a variable with a distinctive name for this value so someone in the future who will read this code will know what it means. It's a part or a CleanCode. 
Another thing is to throw RuntimeException inside curly brackets, It's acceptable to do it this way if it's decided on the project level but the standard in Java is to have it in curly brackets on a separate line. Also you should check the general approach how something like this is being done in the project. The rule is Consistency is more important than best approach.

Reply:
DONE